experimental/dds/tree2/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -260,15 +260,16 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 			commit,
 			sessionId: this.editManager.localSessionId,
 		});
-		this.submitLocalMessage(message);
+		this.submitLocalMessage(this.serializer.encode(message, this.handle));
 	}
 
 	protected processCore(
 		message: ISequencedDocumentMessage,
 		local: boolean,
 		localOpMetadata: unknown,
 	) {
-		const { commit, sessionId } = this.messageCodec.decode(message.contents);
+		const contents = this.serializer.decode(message.contents);
```

Comment:
```suggestion
		const contents: unknown = this.serializer.decode(message.contents);
```

the typing on IFluidSerializer is extremely generous and it's easy to make mistakes. Logic here is sound, but generally try to declare an as-specific-type-as-possible when using return value of IFluidSerializer methods.